### PR TITLE
[FIX] Speed up library

### LINKED ISSUE #646: Speed up library
We should use Pytorch learners to make the fit of `y` (much) more efficient.

### PR DESCRIPTION
**Pull request by GrituX**
There were several instances of copy operations on X when only slice views were needed. I made the changes using masks instead of a series of `np.copy` and `np.delete` operations.

### CODE REVIEW

File: src/hidimstat/_utils/utils.py [hunk 1]
```diff
@@ -37,6 +37,18 @@ def _check_vim_predict_method(method):
         )
 
 
+def _generate_mask(array_size, indexes, select_indices=True):
```
Comment (jpaillard) [inline src/hidimstat/_utils/utils.py]:
```suggestion
def _generate_group_mask(array_size, indexes, complement=False):
```

small suggestions: 
 - title could be more explicit
 - IMO having an argument complement to control whether you return a mask of the group or its complement is clearer. But maybe that's just me.

File: src/hidimstat/_utils/utils.py [hunk 2]
```diff
@@ -37,6 +37,18 @@ def _check_vim_predict_method(method):
...
+    """
+    Generates a mask to select indexes or its opposite.
+    """
+    if select_indices:
+        mask = np.zeros(array_size, dtype=bool)
+    else:
+        mask = np.ones(array_size, dtype=bool)
```
Comment (jpaillard) [inline src/hidimstat/_utils/utils.py]: I think you could return `~mask` instead of having an if/else

File: src/hidimstat/knockoffs.py [hunk 3]
```diff
@@ -195,7 +192,7 @@ def fit(self, X, y):
             )
         else:
             self.estimator_ = clone(self.estimator)
-        self.estimator_ = seed_estimator(self.estimator_, self.random_state)
+        self.estimator = seed_estimator(self.estimator, self.random_state)
```
Comment (jpaillard) [inline src/hidimstat/knockoffs.py]: To be scikit-learn compatible, we should not modify the attributes set in the constructor. 
Also, that would not seed `self.estimator_` which is the one used.

**Review by jpaillard — COMMENTED** [review]
This is a good point, thx.

File: src/hidimstat/_utils/utils.py (hunk 1 above)
Comment (GrituX) [reply src/hidimstat/_utils/utils.py]: No you're right. I was struggling to find a naming that made sense, but yours is good. Thx

**Review by jpaillard — APPROVED** [review]
